src/Common/ZooKeeper/ZooKeeperCommon.cpp

```diff
@@ -285,12 +286,20 @@ void ZooKeeperListRequest::writeImpl(WriteBuffer & out) const
 {
     Coordination::write(path, out);
     Coordination::write(has_watch, out);
+    Coordination::write(static_cast<int32_t>(list_request_type), out);
```

Comment:
Why 32 bit? Enum is 8 bit...

Reply:
I based it on OpNum and that ZooKeeper doesn't support reading something which isn't `int32_t`. But now it hit me that zookeeper won't read this at all so we are good...
I'll change it to `uint8_t`.